src/Workspaces/Core/Portable/ExtensionManager/IExtensionManagerExtensions.cs

```diff
@@ -94,36 +94,47 @@ public static void PerformAction(this IExtensionManager extensionManager, object
     public static Func<SyntaxNode, ImmutableArray<TExtension>> CreateNodeExtensionGetter<TExtension>(
         this IExtensionManager extensionManager, IEnumerable<TExtension> extensions, Func<TExtension, ImmutableArray<Type>> nodeTypeGetter)
     {
-        var map = new ConcurrentDictionary<Type, ImmutableArray<TExtension>>();
+        var map = new Dictionary<Type, ImmutableArray<TExtension>>();
 
-        Func<Type, ImmutableArray<TExtension>> getExtensions = (Type t1) =>
+        foreach (var extension in extensions)
```

Comment:
>extensions [](http://example.com/codeflow?start=34&length=10)

I like it. Good observation that allowing it to not handle the subclass could simplify it like this. 

The only concern would be the fragility when new types are added that need to be added to SyntaxNodeTypes, but I'm guessing that doesn't happen often anymore.